dask_ndmeasure/__init__.py

```diff
@@ -70,6 +72,98 @@ def center_of_mass(input, labels=None, index=None):
     return com_lbl
 
 
+def labeled_comprehension(input,
+                          labels,
+                          index,
+                          func,
+                          out_dtype,
+                          default,
+                          pass_positions=False):
+    """
+    Roughly equivalent to [func(input[labels == i]) for i in index].
+
+    Sequentially applies an arbitrary function (that works on array_like input)
+    to subsets of an n-D image array specified by `labels` and `index`.
+    The option exists to provide the function with positional parameters as the
+    second argument.
+
+    Parameters
+    ----------
+    input : array_like
+        Data from which to select `labels` to process.
+    labels : array_like or None
+        Labels to objects in `input`.
+        If not None, array must be same shape as `input`.
+        If None, `func` is applied to raveled `input`.
+    index : int, sequence of ints or None
+        Subset of `labels` to which to apply `func`.
+        If a scalar, a single value is returned.
+        If None, `func` is applied to all non-zero values of `labels`.
+    func : callable
+        Python function to apply to `labels` from `input`.
+    out_dtype : dtype
+        Dtype to use for `result`.
+    default : int, float or None
+        Default return value when a element of `index` does not exist
+        in `labels`.
+    pass_positions : bool, optional
+        If True, pass linear indices to `func` as a second argument.
+        Default is False.
+
+    Returns
+    -------
+    result : ndarray
+        Result of applying `func` to each of `labels` to `input` in `index`.
+    """
+
+    input, labels, index = _utils._norm_input_labels_index(
+        input, labels, index
+    )
+    out_dtype = numpy.dtype(out_dtype)
+    default = numpy.array(default, dtype=out_dtype)[()]
+    pass_positions = bool(pass_positions)
+
+    indices = _compat._indices(
+        input.shape, dtype=numpy.int64, chunks=input.chunks
+    )
+    indices = list(indices)
+    for i in _pycompat.irange(len(indices)):
+        indices[i] *= int(numpy.prod(indices[i].shape[i + 1:]))
+    indices = dask.array.stack(indices).sum(axis=0)
```

Comment:
Creating such a utility function in PR ( https://github.com/dask-image/dask-ndmeasure/pull/31 ).